routes/commentRoutes.js

```diff
@@ -1,5 +1,26 @@
 const express = require('express');
+const commentController = require('../controllers/commentController');
+const authMiddleware = require('../middlewares/userAuth');
 
 const router = express.Router();
 
+router.post(
+  '/:slug',
+  authMiddleware.isAuthenticated,
+  authMiddleware.isAuthorized,
+  commentController.postComment
+);
+router.put(
+  '/:id',
+  authMiddleware.isAuthenticated,
+  authMiddleware.isAuthorized,
+  commentController.editComment
+);
+router.delete(
+  '/:id',
+  authMiddleware.isAuthenticated,
+  authMiddleware.isAuthorized,
```

Comment:
do you want the middleware to handle authorization?
I am still thinking what's the best way to do it. should it be in middleware or should it be in the controller itself...

if we have it in the middleware, then we will need different authorization logic in the file. eg
- for Contributor: authorized user = project owner
- for Comment: authorized user = comment poster
- for User: authorized user = the owner of that profile